### PR TITLE
remove IRState.falseBlock

### PR DESCRIPTION
**Pull request by WalterBright**
Shouldn't need it anymore.

### CODE REVIEW

File: compiler/src/dmd/s2ir.d [hunk 1]
```diff
@@ -152,13 +152,8 @@ private extern (C++) class S2irVisitor : Visitor
...
         {
-            bool ctfe = s.isIfCtfeBlock();         // __ctfe is always false at runtime
-            const isFalse = ctfe;
-
-            irs.falseBlock += isFalse;
-            Statement_toIR(s.ifbody, irs, &mystate);
-            if (isFalse && irs.falseBlock)
-                --irs.falseBlock;
+            if (!s.isIfCtfeBlock())         // __ctfe is always false at runtime
+                Statement_toIR(s.ifbody, irs, &mystate);
```
Comment (ibuclaw) [inline compiler/src/dmd/s2ir.d:156]: Isn't the inverse needed for the else block when `if (!__ctfe) `?

Comment (WalterBright) [reply compiler/src/dmd/s2ir.d:156]: Yes, if we really wanted to support that. I'm on the fence about it. There are other __ctfe constructs, too. I figure we make this solid first.

**Review by ibuclaw — CHANGES_REQUESTED** [review]
This should target `stable`, as it's a continuation of the fix for regression 23710 in all but name.

**Comment by WalterBright** [comment]
```
macOS 13.x x64 (M1), DMD (latest) Failing after 22m
```
Looks like a problem with master, not this PR. @RazvanN7 ?

**Comment by ibuclaw** [comment]
> ```
> macOS 13.x x64 (M1), DMD (latest) Failing after 22m
> ```
> 
> Looks like a problem with master, not this PR. @RazvanN7 ?

Seems like dmd's yl2xp1 intrinsic doesn't work on macOS 13.x (emulating x86 on M1) with small inputs.

```
log1p(0x1p-16382): expected: 0x1p-16382 | got: 0x0p+0
****** FAIL release64 std.math.exponential
core.exception.AssertError@std/math/exponential.d(3765): Assertion failure
----------------

fabs(0x1p-16382) = 3.3621e-4932
fabs(0x1p-16382) <= 0.25 = true
yl2xp1(0x1p-16382) = 0x0p+0
```

**Comment by WalterBright** [comment]
@ibuclaw I made this for stable https://github.com/dlang/dmd/pull/14946 because my dev work on master needs this one.

**Comment by ibuclaw** [comment]
Shouldn't be required to have both, just merge stable then create a PR to `git merge stable --no-ff` on master.

**Comment by WalterBright** [comment]
Every time I've tried something like that I get a PR with ten thousand lines of difference. Just merge both PRs and things will be fine!

**Comment by ibuclaw** [comment]
Automated merging of stable fixes into master is something that needs looking into as it's quite silly to have bug fixes diverged from mainline for any lengthy period of time.

**Comment by WalterBright** [comment]
If git had a `git undo` command I'd be a lot more adventurous with it. But as it stands, it's easier to just delete 16 lines into a new PR and submit it.

**Comment by ibuclaw** [comment]
Can't do anything about Apple having buggy emulators except turn off fyl2x instructions on macOS.

https://github.com/dlang/phobos/pull/8705

**Comment by WalterBright** [comment]
Thanks for identifying the problem and supplying a workaround.

**Comment by ibuclaw** [comment]
Superseded by #14947.

**Comment by ibuclaw** [comment]
Other PR merged.

**Pull request closed without merging**